### PR TITLE
io_expander: Added io_expander of HT8574. (BSP-280)

### PR DESCRIPTION
**Pull request by espressif2022**
# Checklist for new Board Support package or Component

- [ ] Component contains License
- [ ] Component contains README.md
- [ ] Project [README.md](../README.md) updated
- [ ] Component contains idf_component.yml file with `url` field defined
- [ ] Component was added to CI [upload job](https://github.com/espressif/esp-bsp/blob/master/.github/workflows/upload_component.yml#L17)
- [ ] New files were added to CI build job
- [ ] _Optional:_ Component contains unit tests
- [ ] CI passing

# Change description
Added io_expander of HT8574.

### CODE REVIEW

**Comment by tore-espressif** [comment]
@espressif2022 I'm sorry for the very late response, this must have slipped our attention, I will review ASAP

File: components/io_expander/esp_io_expander_ht8574/esp_io_expander_ht8574.c [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+    *value = ht8574->regs.output;
+    return ESP_OK;
+}
+
+static esp_err_t write_direction_reg(esp_io_expander_handle_t handle, uint32_t value)
+{
+    esp_io_expander_ht8574_t *ht8574 = (esp_io_expander_ht8574_t *)__containerof(handle, esp_io_expander_ht8574_t, base);
+    value &= 0xff;
+    ht8574->regs.direction = value;
```
Comment (tore-espressif) [inline components/io_expander/esp_io_expander_ht8574/esp_io_expander_ht8574.c:122]: How does this even work without any write to I2C bus?

File: components/io_expander/esp_io_expander_ht8574/esp_io_expander_ht8574.c [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+static esp_err_t read_direction_reg(esp_io_expander_handle_t handle, uint32_t *value)
+{
+    esp_io_expander_ht8574_t *ht8574 = (esp_io_expander_ht8574_t *)__containerof(handle, esp_io_expander_ht8574_t, base);
+
+    *value = ht8574->regs.direction;
+    return ESP_OK;
+}
+
+static esp_err_t reset(esp_io_expander_t *handle)
```
[the remark this answers is not in the document]
Comment (espressif2022) [reply components/io_expander/esp_io_expander_ht8574/esp_io_expander_ht8574.c:134]: This extended IO is a quasi-bidirectional I/O, there's no INPUT/OUTPUT register. In order to unify the interface, read/write_direction_reg is a status flag here.

[the remark this answers is not in the document]
Comment (tore-espressif) [reply components/io_expander/esp_io_expander_ht8574/esp_io_expander_ht8574.c:134]: I see know, thank you!

**Review by tore-espressif — APPROVED** [review]

**Review by tore-espressif — APPROVED** [review]

**Pull request merged**